### PR TITLE
Highlight stack trace elements from the current project in Test suite reports

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

If ANSI coloring is enabled, stack trace entries corresponding to the currently running project will be highlighted in bold:
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
This is a feature I have wanted to have for a very long time.

Very often I want to be able to distinguish the standard library code from the _tests_ code. Both are often quite intertwined and it is not always that easy to find the stack frames coming from the test file which often offer useful context to find out in which part of a test unit the failure originates.

Having the current project (test suite files) highlighted in bold makes it much easier to distinguish these.

---

I was trying to find some terminal plugins that could highlight the lines for me based on a regex, but I could not find it.

After all, I decided that since our test library is already using ANSI codes to highlight test results, it may be best to implement that directly in Enso.

**Review by GregoryTravis — APPROVED** [review]
This is amazing.

File: distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso [hunk 1]
```diff
@@ -18,7 +19,7 @@ find_project_root path =
 ## PRIVATE
 find_caller_script : Vector Stack_Trace_Element -> File
 find_caller_script stack =
-    find_main idx =
+    find_main idx = if idx >= stack.length then Error.throw (Illegal_State.Error "Could not find `main` method.") else
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso:22]: What's the purpose of this `if`? Trading `IllegalState.Error` for `IndexOutOfBounds.Error`?

File: distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso [hunk 2]
```diff
@@ -39,27 +40,30 @@ type Suite_Config
        Construct a configuration
 
        Arguments:
+       - print_only_failures: Should only failures be printed.
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso:43]: Printing up to ten of failed test cases at the end of test report would be useful!

We have a lot of tests and when I get the final summary (_n failed tests_) I still need to find out which of them failed. Often searching terminal history for `[FAILED]` doesn't work as the history isn't long enough. Then I need to re-run with `... | tee failed.txt` and that slows everything down.

File: distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso [hunk 3]
```diff
@@ -146,3 +156,21 @@ print_group_report group_name test_results config builder =
 escape_xml : Text -> Text
 escape_xml input =
     input.replace '&' '&amp;' . replace '"' '&quot;' . replace "'" '&apos;' . replace '<' '&lt;' . replace '>' '&gt;'
+
+## PRIVATE
+decorate_stack_trace_text (project_root : File) (decorator : Text -> Text) (stack_trace_text : Text) -> Text =
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso]: Can we move this helper method into its own `private` module?

**Review by JaroslavTulach — APPROVED** [review]
Looks useful.

File: distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso (hunk 3 above)
Comment (radeusgd) [reply distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso]: Ok

File: distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso (hunk 2 above)
Comment (radeusgd) [reply distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso:43]: Good suggestion, but unrelated to this PR. We may implement this as a separate change if we get round to it.

**Review by Akirathan — APPROVED** [review]

File: distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso:22]: Yes.

`Could not find main method.` error is **much clearer** than random `The index N is out of bounds in a collection of length M.`

File: distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso (hunk 2 above)
Comment (radeusgd) [reply distribution/lib/Standard/Test/0.0.0-dev/src/Suite_Config.enso:43]: I created a ticket #9729

**Pull request merged**